packages/yoroi-extension/chrome/extension/background/handlers/content/rpc.js

```diff
@@ -0,0 +1,731 @@
+  connectorGetCollateralUtxos,
+  connectorGetUnusedAddresses,
+  connectorGetUsedAddressesWithPaginate,
+  connectorGetUtxosCardano,
+  connectorRecordSubmittedCardanoTransaction,
+  connectorSendTxCardano,
+  connectorGetAssets,
+  getTokenMetadataFromIds,
+  MAX_COLLATERAL,
+  connectorGetDRepKey,
+  connectorGetStakeKey,
+  transformCardanoUtxos,
+  assetToRustMultiasset,
+} from '../../../connector/api';
+import { authSignHexPayload } from '../../../../../app/connector/api';
+import {
+  getCardanoHaskellBaseConfig,
+} from '../../../../../app/api/ada/lib/storage/database/prepackaged/networks';
+import {
+  sendToInjector,
+  getBoundsForTabWindow,
+  popupProps,
+} from './utils';
+import { asGetAllUtxos, } from '../../../../../app/api/ada/lib/storage/models/PublicDeriver/traits';
+import { asAddressedUtxo as asAddressedUtxoCardano, } from '../../../../../app/api/ada/transactions/utils';
+import {
+  getConnectedSite,
+  setConnectedSite,
+  findWhitelistedConnection,
+  connectError,
+  getConnectedWallet,
+  type ConnectedSite,
+  type SignContinuationDataType,
+} from './connect';
+import type { CardanoTxRequest } from '../../../../../app/api/ada';
+import type { NFTMetadata } from '../../../../../app/api/ada/lib/storage/database/primitives/tables';
+
+declare var chrome;
+
+async function confirmSign(
+  tabId: number,
+  request: PendingSignData,
+  connectedSite: ConnectedSite,
+  continuationData: SignContinuationDataType,
+  uid: number,
+): Promise<void> {
+  const bounds = await getBoundsForTabWindow(tabId);
+
+  connectedSite.pendingSigns[String(request.uid)] = {
+    request,
+    openedWindow: false,
+    continuationData,
+    uid,
+  };
+  await setConnectedSite(tabId, connectedSite);
+  chrome.windows.create({
+    ...popupProps,
+    url: chrome.runtime.getURL(`/main_window_connector.html#/signin-transaction`),
+    left: (bounds.width + bounds.positionX) - popupProps.width,
+    top: bounds.positionY + 80,
+  });
+}
+
+declare function _MaybeWallet(cond: true): PublicDeriver<>;
+// eslint-disable-next-line no-redeclare
+declare function _MaybeWallet(cond: false): void;
+
+type RequestType<NeedConnectedWallet, ParamT> = {|
+  wallet: $Call<typeof _MaybeWallet, NeedConnectedWallet>,
+  tabId: number,
+  message: {|
+    uid: number,
+    params: ParamT,
+    url: string,
+    returnType: string,
+  |},
+|};
+
+type ReturnType<ReturnT> = {|
+  ok: ReturnT
+|} | {|
+  err: string
+|} | {|
+  err: {|
+    code: number,
+    info: string,
+  |}
+|} | void;
+
+type HandleFuncType<NeedConnectedWallet, ParamT, ReturnT> = (
+  request: RequestType<NeedConnectedWallet, ParamT>
+) => Promise<ReturnType<ReturnT>>;
+
+type Handler<NeedConnectedWallet, ParamT, ReturnT> = {|
+  needConnectedWallet: NeedConnectedWallet,
+  syncConnectedWallet: boolean,
+  handle: HandleFuncType<NeedConnectedWallet, ParamT, ReturnT>,
+|};
+
+class _TypeCarrier<ParamT, ReturnT> {
+  toBe<NeedConnectedWallet>(
+    needConnectedWallet: NeedConnectedWallet,
```

Comment:
I can't find a case when the `NeedConnectedWallet` generic type does not end-up being a boolean. I think it can be replaced with `boolean` in the parameters here similar to `syncConnectedWallet` and then also completely removed from the generics in the `Handler`, the `HandleFuncType`, and the `RequestType`.

Is it added only for readability? Maybe just something like this?
```
wallet: $Call<typeof _MaybeWallet, boolean /* NeedConnectedWallet */>,
```